test/unit/event/src/EventTest.cpp

```diff
@@ -150,46 +150,56 @@ TEMPLATE_LIST_TEST_CASE("eventReEnqueueShouldBePossibleIfSomeoneWaitsFor", "[eve
             alpaka::test::EventHostManualTrigger<Dev> k1(f1.m_dev);
             alpaka::test::EventHostManualTrigger<Dev> k2(f1.m_dev);
 
-            // q1 = [k1]
             alpaka::enqueue(q1, k1);
+
+            // q1 = [k1]
             REQUIRE(!alpaka::isComplete(k1));
 
-            // q1 = [k1, e1]
             alpaka::enqueue(q1, e1);
+
+            // q1 = [k1, e1]
             REQUIRE(!alpaka::isComplete(k1));
             REQUIRE(!alpaka::isComplete(e1));
 
-            // q1 = [k1, e1, k2]
             alpaka::enqueue(q1, k2);
+
+            // q1 = [k1, e1, k2]
             REQUIRE(!alpaka::isComplete(k1));
             REQUIRE(!alpaka::isComplete(e1));
             REQUIRE(!alpaka::isComplete(k2));
 
             // wait for e1
-            // q2 = [->e1]
             alpaka::wait(q2, e1);
+            // q2 = [->e1]
 
-            // q2 = [->e1, e2]
             alpaka::enqueue(q2, e2);
+
+            // q2 = [->e1, e2]
             REQUIRE(!alpaka::isComplete(e2));
 
             // re-enqueue should be possible
-            // q1 = [k1, e1-old, k2, e1]
             alpaka::enqueue(q1, e1);
+
+            // q1 = [k1, e1, k2, e1_new]
+            // q2 = [->e1, e2]
             REQUIRE(!alpaka::isComplete(k1));
             REQUIRE(!alpaka::isComplete(k2));
             REQUIRE(!alpaka::isComplete(e1));
             REQUIRE(!alpaka::isComplete(e2));
 
-            // q1 = [k2, e1]
             k1.trigger();
+
+            // q1 = [k2, e1_new]
+            // q2 = []
             REQUIRE(alpaka::isComplete(k1));
             REQUIRE(!alpaka::isComplete(k2));
             REQUIRE(!alpaka::isComplete(e1));
-            REQUIRE(!alpaka::isComplete(e2));
+            REQUIRE(alpaka::isComplete(e2));
```

Comment:
This was wrong before. It was working in CUDA because there is an event state `event not ready`. Due to the fact, that's CUDA driver was not updating the state of the event e2 fast enough the event was in the state `event not ready` what we in alpaka interpret as it is not complete (and this is correct and documented).
For testing we need relatable states and no false negatives, therefore `trigger()` is now waiting long enough to update the internal state of alpaka and CUDA objects.